tls/s2n_handshake_type.c

```diff
@@ -56,6 +56,7 @@ S2N_RESULT s2n_handshake_type_set_tls13_flag(struct s2n_connection *conn, s2n_tl
     RESULT_ENSURE_REF(conn);
     RESULT_ENSURE(s2n_connection_get_protocol_version(conn) >= S2N_TLS13, S2N_ERR_HANDSHAKE_STATE);
     conn->handshake.handshake_type |= flag;
+    conn->is_tls13_state_machine = true;
```

Comment:
What happens if a TLS 1.3 flag is set before the server hello, like with early data, but then the server negotiates TLS 1.2 in the server hello? Or is this not possible?

Reply:
I don't exactly know if this set up is possible, but we are only enforce that the state machine hasn't changed after the server hello has been read. So If we've negotiated TLS1.2 it will technically get overwritten in the server hello. I may look into adding a test for this case...

Reply:
Actually this is definitely not possible. If you try to use early data you can't negotiate down to TLS12.